core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -307,4 +360,42 @@ private void sortAttachments(List<AttachmentContextEntity> attachments) {
                                                                        attachment1.getAttachedDate()));
   }
 
+
+  /**
+   * Gets the parent folder node.
+   *
+   * @param workspaceName the workspace name
+   * @param session jcr session
+   * @param driverName the driver name
+   * @param currentFolder the current folder
+   *
+   * @return the parent folder node
+   *
+   * @throws Exception the exception
+   */
+  private Node getParentFolderNode(String workspaceName, Session session, String driverName, String currentFolder) throws Exception {
+    DriveData driveData = manageDriveService.getDriveByName(driverName);
+    StringBuilder parentPath = new StringBuilder("");
+    if(driveData != null) {
+      parentPath.append(driveData.getHomePath());
+    }
+    NodeHierarchyCreator nodeHierarchyCreator = WCMCoreUtils.getService(NodeHierarchyCreator.class);
```

Comment:
Use `ExoContainerContext.getService` instead. Why not adding this service as dependency declared in constructor ?

Reply:
missed it when refactoring the old code.